src/Datadog.Trace/Agent/Jaeger/JaegerOptions.cs

```diff
@@ -31,5 +34,24 @@ public class JaegerOptions
         /// Gets or sets the network stack exporter. Default exporter: JaegerUdpClient.
         /// </summary>
         internal IJaegerClient TransportClient { get; set; } = new JaegerUdpClient();
+
+        /// <summary>
+        /// Creates an instance of <see cref="JaegerOptions"/> based on passed <see cref="FromTracerSettings"/>.
+        /// </summary>
+        /// <param name="settings"><see cref="TracerSettings"/> to read configuration from</param>
+        /// <returns>New instance of <see cref="JaegerOptions"/></returns>
+        public static JaegerOptions FromTracerSettings(TracerSettings settings)
+        {
+            var agentHost = settings.ConfigurationSource?.GetString(ConfigurationKeys.JaegerExporterAgentHost) ?? "localhost";
+            var agentPort = settings.ConfigurationSource?.GetInt32(ConfigurationKeys.AgentPort) ?? 6831;
+            var agentUri = new Uri(settings.ConfigurationSource?.GetString(ConfigurationKeys.AgentUri) ?? $"http://{agentHost}:{agentPort}");
```

Comment:
this is wrong.
see:
1. https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#jaeger-exporter
2. port description https://www.jaegertracing.io/docs/1.22/getting-started/

we support only UDP

Reply:
resolve